lib/jnpr/junos/decorators.py

```diff
@@ -85,36 +85,54 @@ def ignoreWarnDecorator(function):
     For example::
         dev.rpc.get(ignore_warning=True)
         dev.rpc.get(ignore_warning='vrrp subsystem not running')
-        dev.rpc.get(ignore_warning=['vrrp subsystem not running', 'statement not found'])
+        dev.rpc.get(ignore_warning=['vrrp subsystem not running',
+                                    'statement not found'])
         cu.load(cnf, ignore_warning='statement not found')
 
-    :ignore_warning: It can take take boolean value or string or list of string.
-        if True, it will ignore all warning. If string, it will ignore warning if
-        the statement matches given string. If list of strings, it will try to check
-        if warning statement is from any of the given strings in the list:
+    :ignore_warning: It can take take boolean value or string or list of
+        string. If True, it will ignore all warning. If string, it will
+        ignore warning if the statement matches given string. If list of
+        strings, it will try to check if warning statement is from any of the
+        given strings in the list.
+
+    .. note::
+            When the value of ignore_warning is a string, or list of strings,
+            the string is actually used as a case-insensitive regular
+            expression pattern. If the string contains only alpha-numeric
+            characters, as shown in the above examples, this results in a
+            case-insensitive substring match. However, any regular expression
+            pattern supported by the re library may be used for more
+            complicated match conditions.
     """
     @wraps(function)
     def wrapper(*args, **kwargs):
-        if 'ignore_warning' in kwargs:
-            ignore_warn = kwargs.pop('ignore_warning')
+        ignore_warn = kwargs.pop('ignore_warning', False)
+        if ignore_warn:
             try:
                 result = function(*args, **kwargs)
                 return result
             except RpcError as ex:
                 ex.rpc_xml = JXML.remove_namespaces(ex.rpc_xml)
-                if hasattr(ex, 'rpc_error') and\
-                        ex.rpc_error['severity'] == 'warning':
-                    if ignore_warn is True:
-                        return ex.rpc_xml
-                    elif isinstance(ignore_warn, (str, unicode)):
-                        if re.search(ignore_warn, ex.message, re.I):
-                            return ex.rpc_xml
-                    elif isinstance(ignore_warn, list):
-                        for warn_msg in ignore_warn:
-                            if re.search(warn_msg, ex.message, re.I):
-                                return ex.rpc_xml
-                    raise ex
+                if hasattr(ex, 'errs'):
+                    for err in ex.errs:
+                        if err['severity'] == 'warning':
+                            if isinstance(ignore_warn, (str, unicode)):
+                                if not re.search(ignore_warn, err['message'],
```

Comment:
I think I prefer that we continue to use `re.search` instead of `re.match`.

1) `re.match` only checks at the beginning of each string, not the beginning of each line. Unfortunately, the message returned in the RPC reply sometimes has leading and trailing newlines (`\nstatement not found.\n` rather than just `statement not found.`)

2) `re.search` provides more capability and flexibility. For example, if you only want to match at the start of a line, you can do `ignore_warning='^vrrp subsystem not running'` However, I do think it's important that we properly document the behavior, and that's what I attempted to do with the doc string addition I made in this pull request. Here's the addition:
```
+    .. note::
 +            When the value of ignore_warning is a string, or list of strings,
 +            the string is actually used as a case-insensitive regular
 +            expression pattern. If the string contains only alpha-numeric
 +            characters, as shown in the above examples, this results in a
 +            case-insensitive substring match. However, any regular expression
 +            pattern supported by the re library may be used for more
 +            complicated match conditions.
```